velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -64,6 +66,30 @@ FOLLY_ALWAYS_INLINE void hashIntegral(
   });
 }
 
+template <
+    typename T,
+    typename std::enable_if_t<
+        std::is_same_v<T, float> || std::is_same_v<T, double>,
+        int> = 0>
+FOLLY_ALWAYS_INLINE void hashFloating(
+    const DecodedVector& vector,
+    const SelectivityVector& rows,
+    BufferPtr& hashes) {
+  using IntegralType =
+      std::conditional_t<std::is_same_v<T, float>, int32_t, int64_t>;
+  applyHashFunction(rows, vector, hashes, [&](auto row) {
+    if (std::isnan(vector.valueAt<T>(row))) {
+      if constexpr (std::is_same_v<T, float>) {
+        return hashInteger<IntegralType>(0x7fc00000);
+      } else {
+        return hashInteger<IntegralType>(0x7ff8000000000000L);
+      }
+    } else {
+      return hashInteger<IntegralType>(vector.valueAt<IntegralType>(row));
```

Comment:
+0.0 and -0.0 would still be hashed to different values.  But if that is how Presto Java works, then guess we need to live with it?  Would affect verifier result a bit.

Reply:
Hi @Yuhta, thank you for reviewing!
> +0.0 and -0.0 would still be hashed to different values.

Could you elaborate this a bit?

Reply:
So in floating point numbers +0.0 and -0.0 is different, and will be hashed to different values if we don't special case them.  In `folly::hash` it is special cased, but in Presto Java it seems not.

Reply:
@Yuhta if presto java has same behavior why would it affect verifier results ?

Reply:
Because +0 and -0 are same but they have different checksum